snuba/utils/health_info.py

```diff
@@ -122,15 +121,13 @@ def get_health_info(thorough: Union[bool, str]) -> HealthInfo:
 
 
 def filter_checked_storages() -> List[Storage]:
-    datasets = [get_dataset(name) for name in get_enabled_dataset_names()]
-    entities = itertools.chain(*[dataset.get_all_entities() for dataset in datasets])
-
-    storages: List[Storage] = []
-    for entity in entities:
-        for storage in entity.get_all_storages():
-            if storage.get_readiness_state().value in settings.SUPPORTED_STATES:
-                storages.append(storage)
-    return storages
+    filtered_storages: List[Storage] = []
+    all_storage_keys = get_all_storage_keys()
+    for storage_key in all_storage_keys:
+        storage = get_storage(storage_key)
+        if storage.get_readiness_state().value in settings.SUPPORTED_STATES:
+            filtered_storages.append(storage)
+    return filtered_storages
```

Comment:
so it used to be doing: datasets -> entities -> storages
but now since not all storages are attached to entity, you changed it to use "get_all_storage_keys"?

Reply:
That's correct. This health-check ensures that Snuba is able to access each table in ClickHouse via the storage definitions. Now that an entities are optional, we should just `get_all_storage_keys` directly rather than using the entities to obtain the storages.